prov/shm/src/smr_progress.c

```diff
@@ -519,6 +579,30 @@ static int smr_progress_ipc(struct smr_cmd *cmd, struct ofi_mr **mr,
 		ptr = (char *) (uintptr_t) mr_entry->info.ipc_mapped_addr +
 		      (uintptr_t) cmd->msg.data.ipc_info.offset;
 
+	if (cmd->msg.data.ipc_info.iface == FI_HMEM_ROCR) {
+		struct smr_pend_entry *ipc_entry;
+		struct fi_peer_rx_entry *rx = rx_entry;
+		*total_len = 0;
+		ret = 0;
+		if (!rx_entry) {
+			rx = calloc(sizeof(*rx), 1);
+			if (!rx) {
+				ret = -FI_ENOMEM;
+				goto out;
+			}
+
+			rx->iov = iov;
+			rx->count = iov_count;
+		}
```

Comment:
Why are we allocating an rx_entry every time? Just to save the iov/count? If we enter this if, that probably means we are doing an RMA which gets the iov from the cmd data instead of the rx_entry. The sar_entry struct has an iov/count for that reason. I don't think it is necessary to allocate a new one.